# Business Sales Performance Analytics

## KPI Analysis

### Objective

Calculate key business performance indicators (KPIs) to support executive decision-making and dashboard development.

## 1. Load Dataset

In [3]:
import pandas as pd
import numpy as np
pd.set_option("display.max_columns",None)

df=pd.read_csv("../outputs/cleaned_superstore.csv", parse_dates=["Order Date", "Ship Date"] )

df.head()
              

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,State,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit,Order Year,Order Month,Order Quarter,Shipping Days,Profit Margin %
0,1,CA-2016-152156,2016-11-08,2016-11-11,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,Kentucky,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136,2016,November,4,3,16.00
1,2,CA-2016-152156,2016-11-08,2016-11-11,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,Kentucky,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820,2016,November,4,3,30.00
2,3,CA-2016-138688,2016-06-12,2016-06-16,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,California,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714,2016,June,2,4,47.00
3,4,US-2015-108966,2015-10-11,2015-10-18,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310,2015,October,4,7,-40.00
4,5,US-2015-108966,2015-10-11,2015-10-18,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164,2015,October,4,7,11.25


## 2. Data Verification

In [6]:
print("Dataset shape:", df.shape)

Dataset shape: (9994, 26)


## 3. Total Sales Revenue

In [11]:
total_sales = df["Sales"].sum()
print("Total Sales Revenue: ${:,.2f}".format(total_sales))

Total Sales Revenue: $2,297,200.86


## 4. Total Profit

In [14]:
total_profit = df["Profit"].sum()
print("Total Profit: ${:,.2f}".format(total_profit))

Total Profit: $286,397.02


## 5. Total Orders

In [39]:
total_orders = df["Order ID"].nunique()
print("Total Orders:",total_orders)

Total Orders: 5009


## 6. Average Order Value (AOV)

In [20]:
average_order_value = ( 
    total_sales / total_orders
)
print("Average Order Value: {:,.2f}".format(average_order_value))

Average Order Value: 458.61


## 7. Overall Profit Margin

In [41]:
profit_margin = ( 
    total_profit / total_sales
) * 100
print("Profit Margin: {:,.2f}%".format(profit_margin))

Profit Margin: 12.47%


## 8. Best Performing Category

In [29]:
best_category = ( df.groupby("Category")["Profit"].sum().idxmax() )
print("Best Category:",best_category)

Best Category: Technology


## 9. Best Performing Region

In [43]:
best_region = ( df.groupby("Region")["Profit"].sum().idxmax() )
print("Best Region:",best_region)

Best Region: West


## 10.Best Performing Product

In [37]:
best_product = ( df.groupby("Product Name")["Profit"].sum().idxmax() )
print("Best Product:",best_product)

Best Product: Canon imageCLASS 2200 Advanced Copier


## 11. KPI Summary

The following KPIs provide a high-level overview of business performance and will be used in the Executive Dashboard.

In [52]:
kpi_summary = pd.DataFrame({
    "KPI": [
        "Total Sales Revenue",
        "Total Profit",
        "Total Orders",
        "Average Order Value",
        "Profit Margin (%)",
        "Best Category",
        "Best Region",
        "Best Product"
        ],
    "Value" : [
        round(total_sales, 2),
        round(total_profit, 2),
        total_orders,
        round(average_order_value, 2),
        round(profit_margin, 2),
        best_category,
        best_region,
        best_product
    ]
})

kpi_summary

,KPI,Value
0,Total Sales Revenue,2297200.86
1,Total Profit,286397.02
2,Total Orders,5009
3,Average Order Value,458.61
4,Profit Margin (%),12.47
5,Best Category,Technology
6,Best Region,West
7,Best Product,Canon imageCLASS 2200 Advanced Copier


## 12. Export KPI Summary

In [55]:
kpi_summary.to_csv("../outputs/kpi_summary.csv",index=False)
print("KPI Summary exported sucessfully")

KPI Summary exported sucessfully


## 13. KPI Analysis Conclusion

### KPI Summary

The business generated $2.297 million in revenue and $286,397 in profit across 5,009 orders.

Technology emerged as the most profitable category, while the West region delivered the highest overall performance.

The Canon imageCLASS 2200 Advanced Copier was identified as the top-performing product.

The business maintained an overall profit margin of 12.47%, indicating healthy profitability while highlighting opportunities for improvement through discount optimization and category-specific strategies.

### Next Step

Develop an interactive Power BI dashboard to visualize business performance and support decision-making.